gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/auth/user/impl/UserAuthenticationServiceImpl.java

```diff
@@ -181,13 +190,27 @@ private boolean isIndefinitelyLocked(User user) {
      * @param user Authenticated user
      * @return Completable.complete() or Completable.error(error) if account status is not ok
      */
-    private Completable checkAccountStatus(User user) {
+    private Completable checkAccountStatus(User user, Client client) {
         if (!user.isEnabled()) {
             return Completable.error(new AccountDisabledException("Account is disabled for user " + user.getUsername()));
+        } else if (passwordService.checkAccountPasswordExpiry(user, client, domain)) {
+            return Completable.error( new AccountPasswordExpiredException("Account's password is expired "));
         }
         return Completable.complete();
     }
 
+    private boolean checkAccountPasswordExpiry(User user) {
```

Comment:
Now that we have the method in the PasswordService, this one is not useful anymore